MediaBrowser.Common/Events/EventHelper.cs

```
@@ -59,50 +60,5 @@ public static void QueueEventIfNotNull<T>(EventHandler<T> handler, object sender
                 });
             }
         }
-
-        /// <summary>
-        /// Fires the event.
-        /// </summary>
-        /// <param name="handler">The handler.</param>
-        /// <param name="sender">The sender.</param>
-        /// <param name="args">The <see cref="EventArgs" /> instance containing the event data.</param>
-        /// <param name="logger">The logger.</param>
-        public static void FireEventIfNotNull(EventHandler handler, object sender, EventArgs args, ILogger logger)
-        {
-            if (handler != null)
-            {
-                try
-                {
-                    handler(sender, args);
-                }
-                catch (Exception ex)
-                {
-                    logger.ErrorException("Error in event handler", ex);
```

Your change seem to be changing the logic, as previously an error in the handler was logged but not propagated further.

Are you sure that your fix is fully correct?

---

Wrapping a catch around something isn't a proper fix, not having events throw exceptions is.

---

I agree, in theory. In practice I just want to have progress that is not breaking user experience :)

---

@Bond-009 no response... though I think I can live with the logic change as it should eventually make Jellyfin better - more errors **not** silently eaten means we can faster clean them up.